src/pkmx/lcamera/MainActivity.scala

```diff
@@ -819,14 +827,23 @@ class MainActivity extends SActivity with Observable {
         performCapture = Some { () => performClick() }
         onClick { lcamera() foreach { camera => camera.captureSession() flatMap { _.toOption } match {
           case Some(ps: camera.PhotoSession) =>
-            val time = new Time
-            time.setToNow()
-            val filePathBase = Utils.createPathIfNotExist(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DCIM) + "/Camera/") + time.format("IMG_%Y%m%d_%H%M%S")
             val orientation = orientationVar()
-
             ps.capture(focus(), exposure(), orientation, { (result, jpegImage, rawImage) => {
-              saveJpegFile(s"$filePathBase.jpg", jpegImage)
-              if (saveDng()) { saveDngFile(s"$filePathBase.dng", camera.characteristics, result, rawImage, orientation) }
+              if (isImageCaptureActivity) {
+                val saveUri: Option[Uri] = Option(intent.getExtras
+                  .getParcelable(MediaStore.EXTRA_OUTPUT).asInstanceOf[Uri])
+
+                handleCaptureIntent(saveUri, jpegImage)
+              } else {
+                val time = new Time
+                time.setToNow()
+                val filePathBase = Utils.createPathIfNotExist(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DCIM) + "/Camera/") + time.format("IMG_%Y%m%d_%H%M%S")
```

Comment:
Can you move it outside of the lambda so that the time when the shutter button is pressed will be recorded, rather than when the exposure finishes? Thanks.